### PR TITLE
feat: Portal and Category public APIs

### LINKED ISSUE #4600: Add a public API to return the details of a portal.

### PR DESCRIPTION
**Pull request by tejaswinichile**
# Pull Request Template

## Description

1. Return the portals available on the account
2. Return all the categories belonging to the portal with slug

Fixes #4600 

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

1. Written the test cases to get the information passed.
2. As it is a public API no check for authorization is involved.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/controllers/api/v1/accounts/portals_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class Api::V1::Accounts::PortalsController < Api::V1::Accounts::BaseController
   before_action :fetch_portal, except: [:index, :create]
 
   def index
-    @portals = Current.account.portals
+    @portals = Current.account.portals.active
```
Comment (pranavrajs) [inline app/controllers/api/v1/accounts/portals_controller.rb:5]: In the internal APIs, we should list all the portals, right?

File: app/controllers/public/api/v1/portals_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Public::Api::V1::PortalsController < PublicController
+  before_action :set_portal, except: [:index]
+
+  def index
+    @portals = Portal.all.active
+  end
```
Comment (pranavrajs) [inline app/controllers/public/api/v1/portals_controller.rb:6]: How does this work? If it lists all the portals available in the installation, please remove this end point.

We would need the show end point only.

File: app/controllers/api/v1/accounts/portals_controller.rb (hunk 1 above)
Comment (tejaswinichile) [reply app/controllers/api/v1/accounts/portals_controller.rb:5]: Yes, I forgot to remove this after our discussion yesterday. 👍🏼

File: app/views/public/api/v1/models/_category.json.jbuilder [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
+
+json.related_categories do
+  if category.related_categories.any?
+    json.array! category.related_categories.each do |related_category|
+      json.partial! partial: 'associated_category', category: related_category
+    end
+  end
+end
+
+if category.parent_category.present?
```
Comment (sojan-official) [inline app/views/public/api/v1/models/_category.json.jbuilder:15]: root ?

File: app/views/public/api/v1/models/_category.json.jbuilder [hunk 4]
```diff
@@ -1,5 +1,25 @@
...
+  json.parent_category do
+    json.partial! partial: 'associated_category', category: category.parent_category
+  end
+end
+
+if category.linked_category.present?
```
Comment (sojan-official) [inline app/views/public/api/v1/models/_category.json.jbuilder:21]: we will be taking the change for associated category naming as a separate PR?

**Review by sojan-official — COMMENTED** [review]
Looks good to me. 
Maybe we can also take up the `root` and `associated` naming in this PR.

File: app/views/public/api/v1/models/_category.json.jbuilder (hunk 4 above)
Comment (tejaswinichile) [reply app/views/public/api/v1/models/_category.json.jbuilder:21]: Yes. There is one more article public APIs PR is open, I am adding the changes in that.

File: app/views/public/api/v1/models/_category.json.jbuilder (hunk 3 above)
Comment (tejaswinichile) [reply app/views/public/api/v1/models/_category.json.jbuilder:15]: No no, we have 3 different category associations
1. Parent - sub categories
2. linked_categories which we are going to change to associated categories.
3. related categories with many-to-many associations mostly work like tags.

**Comment by tejaswinichile** [comment]
@sojan-official Can you have a look in this PR again?

**Pull request merged**